app/models/concerns/indexable.rb

```diff
@@ -20,17 +20,11 @@ module Indexable
         end
       end
 
-      if (
-         instance_of?(DataciteDoi) || instance_of?(OtherDoi) ||
-           instance_of?(Doi)
-       ) &&
-          (
-            saved_change_to_attribute?("related_identifiers") ||
-              saved_change_to_attribute?("creators") ||
-              saved_change_to_attribute?("funding_references")
-          )
-        if aasm_state == "findable" && !Rails.env.test?
-          send_import_message(to_jsonapi)
+      if instance_of?(DataciteDoi) || instance_of?(OtherDoi) || instance_of?(Doi)
+        if aasm_state == "findable"
+          changed_attributes = saved_changes
+          relevant_changes = changed_attributes.keys & %w[related_identifiers creators funding_references aasm_state]
+          send_import_message(to_jsonapi) if relevant_changes.any?
```

Comment:
I really like this refactor away from an ever-increasing list of boolean clauses, very elegant!

Reply:
Thank you @digitaldogsbody :)

Also `saved_change_to_attribute?` this method only execute when we have `update` call, so we never captured the `create` DOI call before because of this method checks.

We missed the events for the DOI which are created directly with `findable` state.

Reply:
`saved_changes` will work for both `create` and `update` action.